js/background.js

```diff
@@ -164,4 +182,133 @@ chrome.runtime.onInstalled.addListener((details) => {
+    sessionStorage.setItem('HoverZoomHeaderSettings', JSON.stringify(hoverZoomHeaderSettings));
+}
+
+// update request header(s) just before sending
+function updateRequestHeaders(e) {
+    console.log('updateRequestHeaders');
+    console.log(e);
+
+    let settings = findHeaderSettings(e.url, "request");
+    if (! settings) return;
+
+    // check if update must be skipped because of initiator
+    if (settings.skipInitiator && e.initiator.indexOf(settings.skipInitiator) != -1) {
+        console.log('skip: initiator');
+        return;
+    }
+
+    return { requestHeaders: updateHeaders(e.requestHeaders, settings) };
+}
+
+// update response header(s) just after receiving
+function updateResponseHeaders(e) {
+    console.log('updateResponseHeaders');
+    console.log(e);
+
+    let settings = findHeaderSettings(e.url, "response");
+    if (! settings) return;
+
+    // check if update must be skipped because of initiator
+    if (settings.skipInitiator && e.initiator.indexOf(settings.skipInitiator) != -1) {
+        console.log('skip: initiator');
+        return;
+    }
+
+    return { responseHeaders: updateHeaders(e.responseHeaders, settings) };
+}
+
+// find header(s) setting(s) associated with url that triggered the listener
+function findHeaderSettings(url, requestOrResponse) {
+    let hoverZoomHeaderSettings = sessionStorage.getItem('HoverZoomHeaderSettings') || '{}';
+    hoverZoomHeaderSettings = JSON.parse(hoverZoomHeaderSettings);
+    let reqres = (requestOrResponse == 'request' ? hoverZoomHeaderSettings.requetes || [] : hoverZoomHeaderSettings.responses || []);
+    let settings = reqres.find(s => url.indexOf(s.url) != -1);
+    if (! settings) {
+        console.log('skip: no settings');
+        return null; // no settings found for url
+    }
+    return settings;
+}
+
+// update header(s) according to plug-in settings
+function updateHeaders(headers, settings) {
+    settings.headers.forEach(h => {
+        // types of update:
+        // - 'remove':  remove header
+        // - 'replace': replace header, if header does not exist then do nothing
+        // - 'add':     add header, if header already exists then replace it
+        if (h.typeOfUpdate == 'remove') {
+            let index = headers.findIndex(rh => rh.name.toLowerCase() == h.name.toLowerCase());
+            if (index != -1) headers.splice(index, 1);
+        }
+        if (h.typeOfUpdate == 'replace') {
+            let index = headers.findIndex(rh => rh.name.toLowerCase() == h.name.toLowerCase());
+            if (index != -1) headers[index] = { 'name':h.name, 'value':h.value };
+        }
+        if (h.typeOfUpdate == 'add') {
+            let index = headers.findIndex(rh => rh.name.toLowerCase() == h.name.toLowerCase());
+            if (index != -1) headers[index] = { 'name':h.name, 'value':h.value };
+            else headers.push( { 'name':h.name, 'value':h.value } );
+        }
+    })
+    console.log(headers);
```

Comment:
Can you please change all `console.log` into `cLog` to avoid spamming the console for end users?

Reply:
Oups, done in #932